app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -723,204 +723,265 @@ describe('NetworkController', () => {
   });
 
   describe('getEIP1559Compatibility', () => {
-    describe('when the latest block has a baseFeePerGas property', () => {
-      it('stores the fact that the network supports EIP-1559', async () => {
-        await withController(
-          {
-            state: {
-              networkDetails: {
-                EIPS: {},
-              },
+    describe('if no provider has been set yet', () => {
+      it('does not make any state changes', async () => {
+        await withController(async ({ controller }) => {
+          const promiseForNoStateChanges = waitForStateChanges({
+            controller,
+            count: 0,
+            operation: async () => {
+              await controller.getEIP1559Compatibility();
             },
-          },
-          async ({ controller }) => {
-            const fakeProvider = buildFakeProvider([
-              {
-                request: {
-                  method: 'eth_getBlockByNumber',
-                },
-                response: {
-                  result: POST_1559_BLOCK,
-                },
-              },
-            ]);
-            const fakeNetworkClient = buildFakeClient(fakeProvider);
-            mockCreateNetworkClient().mockReturnValue(fakeNetworkClient);
-            await controller.initializeProvider();
-
-            await controller.getEIP1559Compatibility();
+          });
 
-            expect(controller.store.getState().networkDetails.EIPS[1559]).toBe(
-              true,
-            );
-          },
-        );
+          expect(Boolean(promiseForNoStateChanges)).toBe(true);
+        });
       });
 
-      it('returns true', async () => {
+      it('returns false', async () => {
         await withController(async ({ controller }) => {
-          const fakeProvider = buildFakeProvider([
-            {
-              request: {
-                method: 'eth_getBlockByNumber',
-              },
-              response: {
-                result: POST_1559_BLOCK,
-              },
-            },
-          ]);
-          const fakeNetworkClient = buildFakeClient(fakeProvider);
-          mockCreateNetworkClient().mockReturnValue(fakeNetworkClient);
-          await controller.initializeProvider();
-
-          const supportsEIP1559 = await controller.getEIP1559Compatibility();
+          const isEIP1559Compatible =
+            await controller.getEIP1559Compatibility();
 
-          expect(supportsEIP1559).toBeTruthy();
+          expect(isEIP1559Compatible).toBe(false);
         });
       });
     });
 
-    describe('when the latest block does not have a baseFeePerGas property', () => {
-      it('stores the fact that the network does not support EIP-1559', async () => {
+    describe('if a provider has been set but networkDetails.EIPS in state already has a "1559" property', () => {
```

Comment:
Hmm. These tests seem incorrect; the provider isn't being set.

Reply:
PR created on the core side to fix these tests: https://github.com/MetaMask/core/pull/1420

Reply:
Tests fixed here: [cb7aafe](https://github.com/MetaMask/metamask-extension/pull/19419/commits/cb7aafefc1d0b6cc0b237999c29fd729622c4c52)